File: Python/EulerProblem0002.py
# By considering the terms in the Fibonacci sequence
# whose values do not exceed four million,
# find the sum of the even-valued terms.

import time

def genFib(num1, num2):
	while True:
		fib = num1 + num2
		yield fib
		num1 = num2
		num2 = fib

def fibEven4M():
	generator = genFib(1, 2)
	total = 0
	x = 2
	while x <= 4000000:
		#print x
		if x % 2 == 0:
			total += x
		x = generator.next()
	return total

time_now = time.time()
print 'Total Even Fibonacci < 4M: ' + str(fibEven4M())
print 'Excecution time: ' + str(time.time() - time_now) + 's'

File: JavaScipt/EulerProblem0002.js
/*
 * By considering the terms in the Fibonacci sequence
 * whose values do not exceed four million,
 * find the sum of the even-valued terms.
 */

// The Global Variable
var fibonacci = [];

function fib1(n, before, current) {
	typeof n === 'undefined' ? n = 100 : n = n; 
	typeof before === 'undefined' ? before = 1 : before = before;
	typeof current === 'undefined' ? current = 2 : current = current;

	if (current > n) { return before; }
	else {
		//console.log(current + ' ');
		fibonacci.push(current);
		return before + fib1(n, current, current + before);
	}
}

//console.log('1');
fibonacci.push(1);
var totalFib = fib1(4000000, 1, 2);

// Print fibonacci array
//console.log(fibonacci);

// Find fibonacci < 4M
var evenVal = 0;
for (i = 1; i <= fibonacci.length; i += 3) { evenVal += fibonacci[i]; }
console.log(evenVal);